src/screens/ShareTrip/ShareTrip.tsx

```diff
@@ -0,0 +1,155 @@
+import { RouteProp } from "@react-navigation/core";
+import { StackNavigationProp } from "@react-navigation/stack";
+import React from "react";
+import { useTranslation } from "react-i18next";
+import {
+    Platform,
+    ScrollView,
+    Share,
+    ShareContent,
+    StyleSheet,
+} from "react-native";
+import { Avatar, Button, Header, Text } from "react-native-elements";
+import SvgLogo from "../../components/SvgLogo";
+import RootStackParamList from "../../types/RootStackParamList";
+import { useCreateInvitationMutation } from "./types/create-invite.mutation";
+
+type ShareTripScreenNavigationProp = StackNavigationProp<
+    RootStackParamList,
+    "ShareTrip"
+>;
+type ShareTripScreenRouteProp = RouteProp<RootStackParamList, "ShareTrip">;
+
+type Props = {
+    navigation: ShareTripScreenNavigationProp;
+    route: ShareTripScreenRouteProp;
+};
+
+export default function ShareTrip(props: Props): JSX.Element {
+    const { t } = useTranslation();
+
+    const trip = props.route.params.trip;
+    const [execute, { error, loading }] = useCreateInvitationMutation();
+
+    const invitationBaseUrl = "https://vacatius.com/invite/";
+
+    const handleShare = () => {
+        execute({
+            variables: {
+                input: {
+                    tripId: trip.id,
+                },
+            },
+        })
+            .then((result) => {
+                handleSystemShareSheet(
+                    invitationBaseUrl + result.data?.createInvitation.id
+                );
+            })
+            .catch((error) => console.error(error)); // TODO - Notify user with error modal
+    };
+
+    const handleSystemShareSheet = async (invitationLink: string) => {
+        try {
+            let shareObject: ShareContent = {
+                title: t("androidShareSheetTitle").toString(),
+                message: invitationLink,
+            };
+            // Only use url to properly display shareable content in iOS share sheet
+            if (Platform.OS === "ios") {
+                shareObject = {
+                    url: invitationLink,
+                };
+            }
+
+            await Share.share({ ...shareObject });
```

Comment:
@alexw-at Could you pleave review this PR on your Android device? I've made a change since the last time that we spoke to improve the UX based on which OS one is using. Thanks!